src/main/java/app/controller/admin/UserController.java

```diff
@@ -61,8 +65,16 @@ public String add(Map<String, Object> model) {
 	}
 
 	@RequestMapping(value = "/create", method = RequestMethod.POST)
-	public String add(@ModelAttribute("user") User user) {
-		userService.saveOrUpdate(user);
-		return "redirect:/admin/user/";
+	public String add(@ModelAttribute("userForm") User userForm) {
+		String roleName = null;
+		int roleId = 0;
+		for(Role role: userForm.getRoles()){
+			roleName = role.getName();
+		}
+		if(roleName.equals("ROLE_ADMIN")) roleId = 1;
+		else if(roleName.equals("ROLE_EDITOR")) roleId = 2;
```

Comment:
xử lý logic cho role viết ở service nhé

cơ mà đoạn foreach là sao gái? 1 user có 1 quyền mà, sao lại có for ở đây nhỉ? for ở đây là em lấy cái role sau cùng nữa ah? xem lại chỗ này nhé

Reply:
vì cái userForm.getRoles() là một thuộc class java.util.LinkedHashSet, nó là một Set<Role> nên mình làm vậy để lấy đối tượng đầu ấy. Em dùng ManyToMany ấy chị

Reply:
1 user có nhìu roles và 1 role có nhìu user ah gái?
hay sao em phải dùng manytomany nhỉ